src/document/DocumentManager.ts

```diff
@@ -34,15 +48,90 @@ export class DocumentManager {
     });
     this.model.document.pathlist.addPath("NewPath");
     this.model.document.history.clear();
+    this.setupEventListeners();
+    this.newFile();
+    this.model.uiState.updateWindowTitle();
+  }
+
+  async handleOpenFileEvent(event: Event<unknown>) {
+    let payload = event.payload as OpenFileEventPayload;
+    if (payload.dir === undefined || payload.name === undefined) {
+      toast.error("File load error: non-UTF-8 characters in file path", {
+        containerId: "MENU",
+      });
+    } else {
+      this.model.uiState.setSaveFileName(payload.name);
+      this.model.uiState.setSaveFileDir(payload.dir);
+      this.model.uiState.setIsGradleProject(payload.adjacent_gradle);
+      if (payload.contents !== undefined) {
```

Comment:
Suggesting additional error handling? Maybe a higher scope catch for consistency.